uxarray/helpers.py

```diff
@@ -39,13 +40,37 @@ def determine_file_type(filepath):
                     filepath, mask_and_scale=False)["grid_center_lon"]
                 mesh_filetype = "scrip"
             except KeyError as e:
-                # ugrid with Mesh2
+
+                # check mesh topology and dimension
                 try:
-                    ext_ds = xr.open_dataset(filepath,
-                                             mask_and_scale=False)["Mesh2"]
-                    mesh_filetype = "ugrid"
+                    standard_name = lambda v: v is not None
+                    # getkeys_filter_by_attribute(filepath, attr_name, attr_val)
+                    # return type KeysView
+                    base_dv_nc = list(
+                        xr.open_dataset(
```

Comment:
xarray does cache datasets but it might be cleaner to just use `open_dataset` once like `ext_ds` above.

Reply:
List is for "xr.open_dataset(
                            filepath, mask_and_scale=False).filter_by_attrs(
                                node_coordinates=standard_name).keys()" There could be multiple keys, so taking the first ("[0]") from first element from the list.

Reply:
Thanks. I meant tthat you could
```
ds = xr.open_dataset(filepath, mask_and_scale=False)
```
and then
```
ds.filter_by_attrs(node_coordinates=standard_name).keys()
```
since it gets opened repeatedly (it doesn't actually, because xarray caches the open file)

Reply:
gotcha, thanks.